src/components/intern-app.html

```diff
@@ -1,137 +1,179 @@
-                { id: "010", firstName: 'Mack', lastName: 'Cage', phoneNumber: 6413223232, email: "nick@nickcage.com" }
-              ];
-            }
+     static get properties() {
+       return {
+         userProfiles: {
+           type: Array,
+           value: () => localStorageApi.getUsers()
+         },
+         addUserSelected: {
+           type: Boolean,
+           value: false
+         },
+         isExpanded: {
+            type: Boolean,
+            value: true
           },
-          addUserSelected: {
+         cardsPopulated: {
             type: Boolean,
             value: false
-          }
-        }
-      }
+         }
+       }
+     }
 
-      constructor() {
-        super();
-        this.boundNewUser = this.newUser.bind(this);
-      }
+     constructor() {
+       super();
+       this.boundNewUser = this.newUser.bind(this);
+     }
 
-      connectedCallback() {
-        super.connectedCallback();
-        window.addEventListener("newUser", this.boundNewUser);   
-      }
+     connectedCallback() {
+       super.connectedCallback();
+       window.addEventListener("newUser", this.boundNewUser);
+     }
 
-      disconnectedCallback() {
-        super.disconnectedCallback();
-        window.removeEventListener("newUser", this.boundNewUser); 
-      }
+     disconnectedCallback() {
+       super.disconnectedCallback();
+       window.removeEventListener("newUser", this.boundNewUser);
+     }
 
-      newUser(event) {
-        var newUserProfile = Object.assign({}, event.detail);
-        newUserProfile.id = this.genId();
-        this.pushToArray(this.userProfiles, newUserProfile);
-        this.addUserSelected = false;
-      }
+     deleteAll() {
+      if (window.confirm("Are you sure you want to delete all users?")) {
+        localStorageApi.deleteUsers();
+        cardsPopulated = false;
+        }  
+      } 
 
-      pushToArray(userProfiles, newUserProfile) {
-        var existingIds = userProfiles.map((newUserProfile) => newUserProfile.id);
-        if (!existingIds.includes(newUserProfile.id)) {
-          this.push('userProfiles', newUserProfile);
-        }
-        else {
-          newUserProfile.id = this.genId();
-          this.push('userProfiles', newUserProfile);
-        };
-      };
+     collapseAll() {}
 
-      genId() {
-        return '' + Math.random().toString(9).substr(2, 3);
-      }
+     newUser(event) {
+       const newUserProfile = Object.assign({},event.detail);
+       newUserProfile.id = localStorageApi.genId();
+       this.pushToArray(this.userProfiles, newUserProfile);
+       this.addUserSelected = false;
+       this.cardsPopulated = true;
+     }
 
-      createNewUserProfile(event) {
-        this.addUserSelected = !this.addUserSelected;
-      }
+     pushToArray(userProfiles, newUserProfile) {
+       const existingIds = userProfiles.map(newUserProfile => newUserProfile.id); 
```

Comment:
you've got a naming collision here. you should update the map param to something else.

Reply:
furthermore, what is `existingIds` being used for?

Reply:
apparently nothing lol the on me. I set it up in local storage and didn't take it out of here.